redaxo/src/addons/media_manager/lib/effects/effect_img2img.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/**
+ * @package redaxo\media-manager
+ */
+
+class rex_effect_img2img extends rex_effect_abstract
+{
+    private static $convert_types = [
+        'png',
+        'jpg',
+        'gif',
+        'webp',
+    ];
+
+    private static $convert_to = [
+        'jpg' => [
+            'ext' => 'jpg',
+            'content-type' => 'image/jpeg',
+        ],
+        'png' => [
+            'ext' => 'png',
+            'content-type' => 'image/png',
+        ],
+        'gif' => [
+            'ext' => 'gif',
+            'content-type' => 'image/gif',
+        ],
+        'webp' => [
+            'ext' => 'webp',
+            'content-type' => 'image/webp',
+        ],
+    ];
+
+    private static $convert_tos = ['jpg', 'png', 'gif', 'webp'];
+    private static $convert_to_default = 'jpg';
+    private static $convert_to_default_quality = 80;
+
+    public function execute()
+    {
+        $this->media->asImage();
+        $imageObject = $this->media->getImage();
+        if (null === $imageObject) {
+            return;
+        }
+
+        if (!isset(self::$convert_to[$this->params['convert_to']])) {
+            $convert_to = self::$convert_to[self::$convert_to_default];
+        } else {
+            $convert_to = self::$convert_to[$this->params['convert_to']];
+        }
+
+        $quality = 0;
+        switch ($convert_to['ext']) {
+            case 'jpg':
+                $quality = rex_addon::get('media_manager')->getConfig('jpg_quality');
+                break;
+
+            case 'webp':
+                $quality = rex_addon::get('media_manager')->getConfig('webp_quality');
+                break;
+
+            case 'png':
+                $quality = rex_addon::get('media_manager')->getConfig('png_compression');
+                break;
+        }
+
+        if (!$quality) {
+            $quality = self::$convert_to_default_quality;
+        }
+        if ($this->params['quality'] && (int) $this->params['quality'] > 0) {
+            $quality = $this->params['quality'];
+        }
+        if (100 < (int) $quality) {
+            $quality = 100;
+        }
+        if ('png' == $convert_to['ext']) {
+            if (9 < (int) $quality) {
+                $quality = 9;
+            }
+        }
+
+        $from_path = realpath($this->media->getMediaPath());
+        $ext = rex_file::extension($from_path);
```

Comment:
Der MediaPath muss nicht immer gesetzt sein. Zum Beispiel wenn [das Bild aus dem Feeds-Addon kommt](https://github.com/FriendsOfREDAXO/feeds/blob/84f2314fc9c4e526ed21dadbd857c9485e1e0ae8/lib/media_manager_effect.php#L46).
Daher besser mit `$this->media->getFormat()` arbeiten.

Reply:
Ist im Effekt convert2img auch so gelöst. Sollte man evtl. dann auch gleich anpassen!?
https://github.com/redaxo/redaxo/blob/3b1133b71e548eeb10f5c3ca4354d4797027e700/redaxo/src/addons/media_manager/lib/effects/effect_convert2img.php#L55

Reply:
Ja wäre gut, dort auch anzupassen. Eigentlich am besten als separater PR.
Wenn du möchtest, kannst du es aber auch hier mit anpassen.